pkg/v1/tkg/tkgpackageclient/package_update.go

```diff
@@ -55,31 +59,25 @@ func (p *pkgClient) UpdatePackage(o *tkgpackagedatamodel.PackageOptions, progres
 	}
 
 	pkgInstallToUpdate := pkgInstall.DeepCopy()
-	if o.Version != pkgInstallToUpdate.Status.Version {
-		if pkgInstallToUpdate.Spec.PackageRef == nil || pkgInstallToUpdate.Spec.PackageRef.VersionSelection == nil {
-			err = errors.New(fmt.Sprintf("failed to update package '%s'", o.PkgInstallName))
-			return
-		}
-		progress.ProgressMsg <- fmt.Sprintf("Getting package metadata for '%s'", pkgInstallToUpdate.Spec.PackageRef.RefName)
-		o.PackageName = pkgInstallToUpdate.Spec.PackageRef.RefName
-		if _, _, err = p.GetPackage(o); err != nil {
-			return
-		}
-		pkgInstallToUpdate.Spec.PackageRef.VersionSelection.Constraints = o.Version
+
+	if pkgInstallToUpdate.Spec.PackageRef == nil || pkgInstallToUpdate.Spec.PackageRef.VersionSelection == nil {
```

Comment:
Correct but at the time of installation of the package, ```VersionSelection``` should automatically get filled. Now that we are in update, if it is nil something is really wrong. It can not be nil, but I didn't want to set this variable in line 75 without having made the above check.

Reply:
Got it. We are assuming the package plugin will not work with the pkgi created manually, right? The users could directly apply the yaml of a PKGI that doesn't have versionSelection.